src/components/Sequencer.tsx

```diff
@@ -30,62 +39,82 @@ declare type Props = {
 }
 
 const useStyles = makeStyles((theme: Theme) => ({}));
-
+let beatFunction: any = null;
+let loops: any = {
+  modules: [],
+};
 function Sequencer({ match }: Props) {
   const classes = useStyles();
   const requestRef = useRef<number>(0);
   // @ts-ignore Type '{ state: SchedulerContextType; dispatch: Dispatch<any>; }' is not an array type.
   const [state, dispatch] = useContext(SchedulerContext);
   const [isLoading, setIsLoading] = useState<boolean>(true);
+  let groupedModules;
+
+  const playAudioLogic = useCallback((currentNote: number, globalState: any) => {
+    const currentBeat = indexToBeatKeyMap[currentNote];
+    groupedModules = _.groupBy(globalState.modules, 'instrument');
+    if (groupedModules.bass[0][currentBeat]) {
```

Comment:
right now the key we are using is `'bass drum'` when adding modules. we will need this to match.

Reply:
hmmm not sure how that is even working

Reply:
Understood.  Will need to coordinate the front to match the back.  For the recording functionality, we'll need to figure out the whole dynamic audio stream process instead of the hardcoding approach!